app/controllers/public.js

```diff
@@ -37,4 +37,13 @@ export default class PublicController extends Controller {
   toggleMenu() {
     this.toggleProperty('isMenuOpen');
   }
+
+  @action
+  goToLink(section) {
+    this.transitionToRoute('public.index');
+    setTimeout(function() {
+      document.querySelector(`[href='#${section}']`).click();
+    }, 2000);
```

Comment:
Too much waiting. You should not depend on random waits and timeouts. See if `transitionToRoute` returns a promise and then await it

Reply:
@iamareebjamal now what I found is that I can do 
```javascript
...

transitionToRoute('public.index').promise.then(
   document.querySelector(`[href='#${section}']`).click();
)

... 
```

but this is not working

Reply:
This is not even valid JS syntax. Of course it won't work

Reply:
Make action async and await on transitionToRoute

Reply:
@iamareebjamal  its not working with this also

Reply:
transitionToRoute promise completes before it finished rendering items

Reply:
Then your first solution seems like the best option

Reply:
@iamareebjamal setTimeout one?

Reply:
Yes, that was your first solution

Reply:
@iamareebjamal OK so should I change the time, or 2sec is ok? and if everything else is fine then I think this is done. Let me know anything to change.

Reply:
2 sec was not in your first solution.

> Then your first solution seems like the best option

Read your first solution again

Reply:
So you are talking about my first commit 'Fix Double Click' ?

Reply:
I'm talking about first working solution which was setTimeout without delay

Reply:
so sorry But it was always with 2000ms, I never set setTimeout with 0ms.
![Screenshot from 2020-11-08 03-38-17](https://user-images.githubusercontent.com/71120226/98452301-f7f64680-2173-11eb-8106-5f67ce782bcd.png)

Reply:
@iamareebjamal First solution was for fixing button for same page 
second one was for fixing from different page which is above 
third was improving the above code and using click() handler.
and in second and third settimeout with 2000ms was used. So pls let me know if I am missing something to look at?

Reply:
@iamareebjamal please let me know